api/draft_registrations/serializers.py

```diff
@@ -0,0 +1,215 @@
+    tags = ValuesListField(attr_name='name', child=ser.CharField(), required=False)
+    node_license = NodeLicenseSerializer(required=False, source='license')
+
+    links = LinksField({
+        'self': 'get_self_url',
+    })
+
+    affiliated_institutions = RelationshipField(
+        related_view='draft_registrations:draft-registration-institutions',
+        related_view_kwargs={'draft_id': '<_id>'},
+        self_view='draft_registrations:draft-registration-relationships-institutions',
+        self_view_kwargs={'draft_id': '<_id>'},
+        read_only=False,
+        many=True,
+        required=False,
+    )
+
+    branched_from = NodeRelationshipField(
+        related_view=lambda n: 'draft-nodes:draft-node-detail' if getattr(n, 'type', False) == 'osf.draftnode' else 'nodes:node-detail',
+        related_view_kwargs={'node_id': '<branched_from._id>'},
+        read_only=False,
+        required=False,
+    )
+
+    contributors = RelationshipField(
+        related_view='draft_registrations:draft-registration-contributors',
+        related_view_kwargs={'draft_id': '<_id>'},
+    )
+
+    license = NodeLicenseRelationshipField(
+        related_view='licenses:license-detail',
+        related_view_kwargs={'license_id': '<license.node_license._id>'},
+        read_only=False,
+    )
+
+    @property
+    def subjects_related_view(self):
+        # Overrides TaxonomizableSerializerMixin
+        return 'draft_registrations:draft-registration-subjects'
+
+    @property
+    def subjects_view_kwargs(self):
+        # Overrides TaxonomizableSerializerMixin
+        return {'draft_id': '<_id>'}
+
+    @property
+    def subjects_self_view(self):
+        # Overrides TaxonomizableSerializerMixin
+        return 'draft_registrations:draft-registration-relationships-subjects'
+
+    def get_self_url(self, obj):
+        return absolute_reverse(
+            'draft_registrations:draft-registration-list',
+            kwargs={
+                'version': self.context['request'].parser_context['kwargs']['version'],
+            },
+        )
+
+    # Overrides DraftRegistrationSerializerLegacy
+    def get_node(self, validated_data):
+        # Node comes from branched_from relationship rather than from URL
+        return validated_data.pop('branched_from', None)
+
+    def expect_subjects_as_relationships(self, request):
+        """Determines whether subjects should be serialized as a relationship.
+        Older serializers expect subjects as attributes for earlier versions,
+        but this new serializer does not have to adhere to that same behavior.
+        :param object request: Request object
+        :return bool: Subjects should be serialized as relationships
+        """
+        # Overrides TaxonomizableSerializerMixin
+        return True
+
+
+class DraftRegistrationDetailSerializer(DraftRegistrationSerializer, DraftRegistrationDetailLegacySerializer):
+    """
+    Overrides DraftRegistrationSerializerLegacy to make id required.
```

Comment:
`Overrides DraftRegistrationDetailLegacySerializer...`

Reply:
Add tests to ensure that the id is required

Reply:
The ID would be given in the query parameter in the URL. If theres no ID given, it would just be the draft registration list endpoint.